satpy/readers/gerb_l2_hr_h5.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2023
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+
+"""A reader for the Top of Atmosphere outgoing fluxes from the Geostationary Earth Radiation
+Budget instrument aboard the Meteosat Second Generation satellites."""
+
+
+import logging
+from datetime import timedelta
+
+import dask.array as da
+import h5py
+import numpy as np
+import xarray as xr
+
+from satpy.readers.file_handlers import BaseFileHandler
+from satpy.resample import get_area_def
+
+LOG = logging.getLogger(__name__)
+
+def gerb_get_dataset(hfile, name):
+    """
+    Load a GERB dataset in memory from a HDF5 file
+
+    The routine takes into account the quantisation factor and fill values.
+    """
+    ds = hfile[name]
+    if 'Quantisation Factor' in ds.attrs and 'Unit' in ds.attrs:
+        ds_real = ds[...]*ds.attrs['Quantisation Factor']
+    else:
+        ds_real = ds[...]*1.
+    ds_min = ds[...].min()
+    if ds_min < 0:
+        mask = ds == ds_min
+        ds_real[mask] = np.nan
```

Comment:
I'll mention other ideas for dask-ification in my main review comment, but wanted to point out that this is very inefficient and if/when it is done with dask arrays it will be even more inefficient. It was hard for me to understand the product guide regarding fill values, but it looks like the fill or "Error" value is always the minimum value of the in-file data type. Does that sound right?

In that case, can we hardcode this fill value in the YAML or look it up using something like `np.iinfo(ds.dtype).min`. Or are there attributes in the file that tell us this information? That way we don't have to look at every pixel just to determine the minimum value that we use for masking. Also, why does the minimum value have to be `< 0` to be used for masking?